### DISCUSSION TITLE
WKT-Representation of geometry as Oracle CLOB data type can not be recognized as geometry

### DISCUSSION THREAD

**Discussion by niels-flohr** [dbeaver/dbeaver#22236] [Q&A]
### Description

I have have data stored as SDO- and ST-Geometry in Oracle Databases. For SDO-Geometries the command

`SDO_UTIL.TO_WKTGEOMETRY(SDO_CS.TRANSFORM(SHAPE, 4326)) AS GEOM`

in a SELECT Statement can be used to transform the geometry in a WKT-representation. Oracle returns a CLOB Datatype, but DBeaver can not interpret it correctly as geometry column, only if the column is cast to VARCHAR2 datatype, which is not feasible for more complex geometries requiring more than 2000 Characters. 

In my understanding something similar was already implemented for [BLOB-like binary datatypes](https://github.com/dbeaver/dbeaver/commit/5c677a9dc33c72d1ab57a568c06abd2232560e29), is it possible to provide the same for the [CLOB ](https://docs.oracle.com/javadb/10.10.1.2/ref/rrefclob.html) type?

CLOB: 

![grafik](https://github.com/dbeaver/dbeaver/assets/34693249/65ed98fe-307f-4a0d-beae-750e57cc8e6f)

CAST AS VARCHAR2:

![grafik](https://github.com/dbeaver/dbeaver/assets/34693249/a8c5c224-5468-4ae4-a2bd-cdaeaf9dd35d)

### DBeaver Version

CE 23.3.0.202312122044

### Operating System

Win 10

### Database and driver

Oracle 12 (12.2.0.1.0)
Oracle JDBC 12.2.01

**Comment by LonwoLonwo** [comment]
Hello @niels-flohr 

We have support for GIS data types for Oracle in our PRO Products. You can try it for free: https://dbeaver.com/trial/

![2023-12-21 19_34_16-DBeaver Ultimate 23 3 1 - _orcl_ Script-59](https://github.com/dbeaver/dbeaver/assets/45152336/367606e0-7ad6-4412-9ada-1c3472afded8)

**Reply by niels-flohr** [reply]
Yes, but this does only work for SDO_Geometries not ESRI ST_Geometries in Oracle in my understanding. 
So for quick validation and data transfer the WKT-representations of geometries are necessary and should imho be displayable regardless of licensing - especially as i pointed out, if such a support is provided for binary data / BLOBs on other DBs. Especially since wkt is an [OGC](https://en.wikipedia.org/wiki/Open_Geospatial_Consortium) standard.
